### PR TITLE
EOS-25696: WIP- Partition table support for Partition Stob

### PR DESCRIPTION
**Pull request by vidyadhar-pinglikar**
# Problem Statement
- Partition Table support to use single disk for seg0, seg1, log and data

# Design
-  https://seagate-systems.atlassian.net/wiki/spaces/~91827731/pages/712966428/Partition+Stob+Domain

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- WIP code for early review, tested single read/write on single-node vm.

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
 - https://seagate-systems.atlassian.net/wiki/spaces/~91827731/pages/712966428/Partition+Stob+Domain

### CODE REVIEW

File: motr/magic.h [hunk 1]
```diff
@@ -976,7 +976,13 @@ enum m0_magic_satchel {
 	/* stob/ad.c:ad_domains_tl::td_head_magic (ad sold head) */
 	M0_AD_DOMAINS_HEAD_MAGIC    = 0x33ad501d4ead77,
 
+	/* part_domain_map:pdm_magic (part sold class) */
```
Comment (mukundkanekar) [inline motr/magic.h:979]: need to update to unique value.

File: be/test_partition_table.c [hunk 2]
```diff
@@ -0,0 +1,159 @@
+/*
```
Comment (mukundkanekar) [inline be/test_partition_table.c:1]: This file is no longer needed, will be removed.

File: be/domain.c [hunk 3]
```diff
@@ -102,6 +109,28 @@ M0_INTERNAL int m0_be_segobj_opt_begin(struct m0_be_seg         *dict,
 	return m0_be_segobj_opt_iterate(dict, objtype, opt, suffix, true);
 }
 
+
+// M0_INTERNAL int m0_be_domain_stob_open(struct m0_be_domain  *dom,
```
Comment (mukundkanekar) [inline be/domain.c:113]: need to be removed.

File: be/domain.c [hunk 4]
```diff
@@ -714,19 +756,53 @@ static int be_domain_level_enter(struct m0_module *module)
...
 						   &dom->bd_stob_domain));
-
 	case M0_BE_DOMAIN_LEVEL_NORMAL_STOB_DOMAIN_INIT:
 		if (cfg->bc_mkfs_mode)
 			return M0_RC(0);
 		return M0_RC(m0_stob_domain_init(cfg->bc_stob_domain_location,
 						 cfg->bc_stob_domain_cfg_init,
 						 &dom->bd_stob_domain));
+	case M0_BE_DOMAIN_LEVEL_MKFS_PART_STOB_DOMAIN_DESTROY:
+#if 0
```
Comment (mukundkanekar) [inline be/domain.c:768]: #TODO uncomment and test.

File: be/log_store.c [hunk 5]
```diff
@@ -37,13 +38,13 @@
...
 #include "stob/domain.h"        /* m0_stob_domain_create */
-
+#include "stob/partition.h"
 /**
  * @addtogroup be
  *
  * @{
  */
-
+static const struct m0_modlev be_log_store_levels[];
```
Comment (mukundkanekar) [inline be/log_store.c:47]: Added for debug, to be removed.

File: motr/setup.c [hunk 6]
```diff
@@ -1086,6 +1088,7 @@ static int cs_storage_init(const char *stob_type,
 						     &stob->s_sdom);
 		if (rc != 0)
 			stob->s_sdom = NULL;
+		stob->s_sdom->sd_ad_mode = 1;
```
Comment (mukundkanekar) [inline motr/setup.c:1091]: unused, to be removed.

File: stob/domain.h [hunk 7]
```diff
@@ -103,6 +103,7 @@ struct m0_stob_domain {
 	struct m0_stob_cache		 sd_cache;
 	/** Private data of stob domain implementation */
 	void                            *sd_private;
+	bool                             sd_ad_mode;
```
Comment (mukundkanekar) [inline stob/domain.h:106]: unused, to be removed.

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Comment by shailesh-vaidya** [comment]
Closing as an obsolete

**Pull request closed without merging**